### PR TITLE
Chore/ddw 218 Improve request body params length calculation

### PR DESCRIPTION
**Pull request by nikolaglumac**
This PR improves the request body length calculation as per recommendation by @cleverca22 given in: https://github.com/input-output-hk/daedalus/pull/839#issuecomment-378896197.

### CODE REVIEW

**Comment by KtorZ** [comment]
Where does that `Buffer` comes from in browsers ^.^ ?

**Comment by nikolaglumac** [comment]
@KtorZ in this case it comes from node. Daedalus is in the end Electron app so we have this at our disposal ;)

**Comment by DominikGuzei** [comment]
@nikolaglumac @KtorZ please be aware though that this will soon change since we will disable node integration in this sprint … so it would be better if you find a solution without node integration before we merge this change 😉

**Comment by DominikGuzei** [comment]
@nikolaglumac i actually forgot what the initial reason was for the change? If it was performance we can safely abort this work since this is not a performance critical part of the application.

**Comment by nikolaglumac** [comment]
@DominikGuzei yes the performance was the reason.

**Comment by nikolaglumac** [comment]
@DominikGuzei @KtorZ @cleverca22 there is apparently a "browser" version of node.js Buffer: https://github.com/feross/buffer
Maybe we could use this one? My question is if this would really be an advantage over the current implementation with `(new TextEncoder()).encode(requestBody).length`.
I would like to hear your opinion on this one...

**Comment by KtorZ** [comment]
In JavaScript (or any language in general), I'd rather rely on built-in libraries than pulling in an extra dependency. 
I am not sure that the "performance" argument has to come into play in that as we're probably comparing nanoseconds here. The Buffer interface is definitely more elegant, though, if we move the instantiation of the `TextEncoder` elsewhere, we would also have something quite neat:

```
encoder.encode(requestBody).length
```

File: source/renderer/app/api/ada/lib/request.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ function typedRequest<Response>(
       hasRequestBody = true;
       requestBody = JSON.stringify(rawBodyParams);
       options.headers = {
-        'Content-Length': (new TextEncoder()).encode(requestBody).length,
+        'Content-Length': Buffer.byteLength(requestBody),
```
[the remark this answers is not in the document]
Comment (cleverca22) [reply source/renderer/app/api/ada/lib/request.js]: i'm thinking do a full `var bytes = Buffer(requestBody,"utf-8");` and then send that exact buffer when doing the post, so the post cant use a different encoding

**Comment by nikolaglumac** [comment]
@KtorZ I like your proposal 👍 - I guess that is the best thing we can do at the moment since we will remove node integration from the renderer process...

**Comment by nikolaglumac** [comment]
@DominikGuzei @KtorZ @cleverca22 I have now reverted the code to the old `TextEncoder` solution with the difference that I have extracted this part into a util which is then reused in all of the 3 request files...

**Comment by nikolaglumac** [comment]
@KtorZ in this case I will leave it as is since it is used only for the requests and thus it is placed in api/lib/utils. In case we will need it elsewhere we will rename it and move it into a more global directory.

**Pull request merged**